packages/frontend/src/features/tableCalculation/components/TableCalculationModal.tsx

```diff
@@ -161,7 +168,36 @@ const TableCalculationModal: FC<Props> = ({
                             />
                         </Tabs.Panel>
                     </Tabs>
-
+                    <Tooltip
+                        position="bottom"
+                        withArrow
+                        multiline
+                        maw={400}
+                        withinPortal
+                        label={
+                            'Manually select the type of the result of this SQL table calculation, this will help us to treat this field correctly in filters or results.'
+                        }
+                    >
+                        <Select
+                            label={'Result type'}
+                            id="download-type"
+                            defaultValue={'default'}
+                            value={
+                                form.getInputProps('type').value ?? 'default'
+                            }
```

Comment:
Default will always be treated as `number` in the filters, I believe this can be confusing if I try to create a custom dimension that returns string and I leave the `type` as default as the filter then doesn't allow entering the `string` value:

https://github.com/lightdash/lightdash/assets/22939015/33964443-d7bc-4d0c-9513-a044e15cce40

Suggestion could be to have the `defaultValue` of the type select be number, what do you think?

Reply:
> Default will always be treated as number in the filters
This is the current behaviour

> Suggestion could be to have the defaultValue of the type select be number, what do you think?
Not a bad idea :thinking: @TuringLovesDeathMetal what do you think ?

Reply:
I don't think that's true? I think it's treated as a "value" type when it's selected as a filter, no?

Reply:
it is treated as a `value` on the chart, but in LIghtdash, we default to `numbers` for example, on filters, so it makes sense to not have a default, and use number instead. (undefined becomes number)

Reply:
Unless we want to use `default` in the future for something like `auto detecting the field based on the SQL` or something

Reply:
^Just to write down what we talked about on the call:

Test that if I have an existing table calculation filter that's using a string operator (e.g. `name` is equal to 'blah'), and we switch table calculations to be assumed as numbers by default, then the existing filter doesn't break.

Reply:
Right, so the query keeps working, there was only one bug I had to fix, but now it looks like this. I added  a warning message if you're trying to use a string operation in a numberf filter. 
> Unexpected operator ${type} for number filter type. If you are using a table calculation, please update its result type to string.

I didn't want to add more `backward compatibility` code, so I think it is ok like this for now. 

![Screenshot from 2024-04-24 13-00-00](https://github.com/lightdash/lightdash/assets/1983672/ea064c7a-283a-43f9-a064-d848b862b682)

Reply:
I also removed the `default` in the UI